### ISSUE TITLE
func-names causes visual clutter inside editors

### ISSUE THREAD

**Issue by xsrvmy** [eslint/eslint#12022]
The func-names rule flags the entire function instead of just the declaration, causing visual clutter in editors such as VSCode.

**What rule do you want to change?**
`func-names`

**Does this change cause the rule to produce more or fewer warnings?**
No

**How will the change be implemented? (New option, new default behavior, etc.)?**
Either new option, or change current behaviour.

**Please provide some example code that this change will affect:**

```js
const x = function () {
  return 100;
}
```

**What does the rule currently do for this code?**
Highlights the entire function declaration as a warning.

**What will the rule do after it's changed?**
Highlight only `function () `.

**Are you willing to submit a pull request to implement this change?**
Maybe

**Comment by platinumazure** [comment]
Hi @xsrvmy, thanks for the issue.

I think this is a great idea. I will champion this change.

**Comment by kaicataldo** [comment]
One thing we should account for when we actually implement this is the potential for return types (I think they should Ben highlighted as well).

**Comment by mysticatea** [comment]
For reference, we have a utility, [getFunctionHeadLoc()](https://github.com/eslint/eslint/blob/02d7542cfd0c2e95c2222b1e9e38228f4c19df19/lib/rules/utils/ast-utils.js#L1204).

**Comment by g-plane** [comment]
I think there's no need to add a new option.

**Comment by platinumazure** [comment]
Since I both championed and :+1:'d this issue, I think we need one more team member in order to accept. (I have removed my :+1: to reduce confusion.)

@eslint/eslint-team Could we get one more :+1: for this issue? This is just to adjust the reporting range for editor integrations.

**Comment by aladdin-add** [comment]
here is the 3rd 👍 , labelled accepted.
